src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java

```diff
@@ -681,6 +681,8 @@ protected void validatePreparse() {
      */
     protected abstract PerDimensionProcessor getPerDimensionProcessor();
 
+    protected abstract VectorTransformer getVectorTransformer();
```

Comment:
Can you add java doc for this one? Also, can you update javadoc for getVectorValidator() to say that it is validated before any transform calls?

Reply:
Ack